### DISCUSSION TITLE
Certificate API enable in sunbird RC

### DISCUSSION THREAD

**Discussion by sachinsahu731** [Sunbird-RC/community#279] [General]
Hi team ,
I am on it to setup sunbird RC in local .
Every thing working fine till creating teacher and student scheme and all API is working . [here](https://docs.sunbirdrc.dev/using-the-apis)
But I want to attach certificate after creating entity or teacher and student .How can I do that .
How to enable certificate API in sunbird RC

**Comment by tejash-jl** [comment]
@sachinsahu731 you can follow it here,
https://github.com/tejash-jl/ref-sunbird-certificate-demo/blob/main/sunbird-certificate-demo.ipynb

**Comment by sachinsahu731** [comment]
@tejash-jl 
While creating the schema we are getting error 
`"status":"UNSUCCESSFUL","errmsg":"User is not
 allowed to perform the operation on this entity"`

**Reply by tejash-jl** [reply]
@sachinsahu731 cross-check if the user has an `admin` role. You can assign the role in keycloak.

**Reply by sachinsahu731** [reply]
@tejash-jl I am only seen three roles in RC offline_access , uma_authorization , default-roles-sunbird-rc where user has default-roles-sunbird-rc

**Reply by tejash-jl** [reply]
you can create a role called `admin` and add that to the user.

**Reply by sachinsahu731** [reply]
@tejash-jl  After adding the admin role to the user we still get 401 unauthorize error while creating schema.

**Reply by tejash-jl** [reply]
Can you share the logs of the registry and docker-compose file?

**Reply by sachinsahu731** [reply]
@tejash-jl  
`sunbirdrc-db-1 | 2022-10-18 10:10:34.634 UTC [39] ERROR:  duplicate key value violates unique constraint "sibling_names"
sunbirdrc-db-1 | 2022-10-18 10:10:34.634 UTC [39] DETAIL:  Key (realm_id, parent_group, name)=(sunbird-rc,  , Issuer) already exists.
sunbirdrc-db-1 | 2022-10-18 10:10:34.634 UTC [39] STATEMENT:  insert into public.KEYCLOAK_GROUP (NAME, PARENT_GROUP, REALM_ID, ID) values ($1, $2, $3, $4)
sunbirdrc-rg-1 | 10:10:34.291 [http-nio-8081-exec-5] INFO  d.s.r.c.RegistryEntityController:71 - Inviting entity {"contactDetails":{"name":"","email":"issuerId","mobile":"issuerId"},"schemas":[],"accountDetails":{"userId":"q7@mailinator1.com"},"websiteUrl":"https://picsum.photos/200","logoUrl":"https://picsum.photos/200","name":"Issuer","sectorType":"Education"}
sunbirdrc-rg-1 | 10:10:34.317 [http-nio-8081-exec-5] WARN  d.s.registry.service.ISearchService:66 - Populates SearchQuery for limit/offset(optional): null
sunbirdrc-rg-1 | 10:10:34.326 [http-nio-8081-exec-5] ERROR dev.sunbirdrc.pojos.RequestWrapper:39 - Can't read from http stream. Set body empty
sunbirdrc-rg-1 | 10:10:34.326 [http-nio-8081-exec-5] ERROR dev.sunbirdrc.pojos.APIMessage:44 - Can't read request body
sunbirdrc-rg-1 | 10:10:34.328 [http-nio-8081-exec-5] INFO  d.s.r.helper.EntityStateHelper:65 - Detecting state changes by comparing attestation paths in existing and the updated nodes
sunbirdrc-rg-1 | 10:10:34.339 [http-nio-8081-exec-5] INFO  d.s.keycloak.KeycloakAdminUtil:70 - Creating user with mobile_number : q7@mailinator1.com
sunbirdrc-rg-1 | 10:10:34.929 [http-nio-8081-exec-5] INFO  d.s.keycloak.KeycloakAdminUtil:77 - Response |  Status: 201 | Status Info: Created
sunbirdrc-rg-1 | 10:10:34.930 [http-nio-8081-exec-5] INFO  d.s.keycloak.KeycloakAdminUtil:78 - User ID path/auth/admin/realms/sunbird-rc/users/05e4b16f-47cc-4f30-8a0b-545678d7d4c7
sunbirdrc-rg-1 | 10:10:34.931 [http-nio-8081-exec-5] INFO  d.s.keycloak.KeycloakAdminUtil:80 - User ID : 05e4b16f-47cc-4f30-8a0b-545678d7d4c7
sunbirdrc-rg-1 | 10:10:34.934 [http-nio-8081-exec-5] INFO  d.s.registry.helper.RegistryHelper:240 - Add api: entity type: Issuer and shard propery: none
sunbirdrc-rg-1 | 10:10:34.936 [http-nio-8081-exec-5] INFO  d.s.registry.sink.shard.ShardManager:43 - Activated shard shard1 for attribute value null
sunbirdrc-rg-1 | 10:10:35.307 [http-nio-8081-exec-5] INFO  d.s.registry.helper.RegistryHelper:251 - AddEntity,1-6d27f8ba-80ff-4eb6-967e-bdff5060d295
sunbirdrc-rg-1 | 10:12:48.703 [http-nio-8081-exec-7] INFO  d.s.r.c.RegistryEntityController:222 - MODE: false
sunbirdrc-rg-1 | 10:12:48.704 [http-nio-8081-exec-7] INFO  d.s.r.c.RegistryEntityController:223 - MODE: null
sunbirdrc-rg-1 | 10:12:48.705 [http-nio-8081-exec-7] INFO  d.s.r.c.RegistryEntityController:224 - Adding entity {"name":"schema21","schema":"{\n  \"$schema\": \"http://json-schema.org/draft-07/schema\",\n  \"type\": \"object\",\n  \"properties\": {\n    \"TrainingCertificate\": {\n      \"$ref\": \"#/definitions/TrainingCertificate\"\n    }\n  },\n  \"required\": [\n    \"TrainingCertificate\"\n  ],\n  \"title\": \"TrainingCertificate\",\n  \"definitions\": {\n    \"TrainingCertificate\": {\n      \"$id\": \"#/properties/TrainingCertificate\",\n      \"type\": \"object\",\n      \"title\": \"The TrainingCertificate Schema\",\n      \"required\": [\n        \"name\",\n        \"contact\"\n      ],\n      \"properties\": {\n        \"name\": {\n          \"type\": \"string\"\n        },\n        \"trainingTitle\": {\n          \"type\": \"string\"\n        },\n        \"contact\": {\n          \"type\": \"string\"\n        },\n        \"date\": {\n          \"type\": \"string\",\n          \"format\": \"date\"\n        },\n        \"note\": {\n          \"type\": \"string\"\n        }\n      }\n    }\n  },\n  \"_osConfig\": {\n    \"uniqueIndexFields\": [\n      \"contact\"\n    ],\n    \"ownershipAttributes\": [],\n    \"roles\": [],\n    \"inviteRoles\": [\n      \"anonymous\"\n    ],\n    \"enableLogin\": false,\n    \"credentialTemplate\": {\n      \"@context\": [\n        \"https://www.w3.org/2018/credentials/v1\",\n        \"https://gist.githubusercontent.com/dileepbapat/eb932596a70f75016411cc871113a789/raw/498e5af1d94784f114b32c1ab827f951a8a24def/skill\"\n      ],\n      \"type\": [\n        \"VerifiableCredential\"\n      ],\n      \"issuanceDate\": \"2021-08-27T10:57:57.237Z\",\n      \"credentialSubject\": {\n        \"type\": \"Person\",\n        \"name\": \"{{name}}\",\n        \"trainedOn\": \"{{trainingTitle}}\"\n      },\n      \"issuer\": \"did:web:sunbirdrc.dev/vc/skill\"\n    },\n    \"certificateTemplates\": {\n      \"html\": \"https://raw.githubusercontent.com/dileepbapat/ref-sunbirdrc-certificate/main/schemas/templates/TrainingCertificate.html\",\n      \"svg\": \"https://raw.githubusercontent.com/dileepbapat/ref-sunbirdrc-certificate/main/schemas/templates/TrainingCertificate.svg\"\n    }\n  }\n}"}
sunbirdrc-rg-1 | 10:12:48.710 [http-nio-8081-exec-7] ERROR d.s.r.c.RegistryEntityController:250 - Exception in controller while adding entity !
sunbirdrc-rg-1 | dev.sunbirdrc.registry.exception.UnAuthorizedException: User is not allowed to perform the operation on this entity
sunbirdrc-rg-1 | at dev.sunbirdrc.registry.helper.RegistryHelper.authorizeUserRole(RegistryHelper.java:932)
sunbirdrc-rg-1 | at dev.sunbirdrc.registry.helper.RegistryHelper.authorizeManageEntity(RegistryHelper.java:907)
sunbirdrc-rg-1 | at dev.sunbirdrc.registry.helper.RegistryHelper$$FastClassBySpringCGLIB$$9ae85a6b.invoke(<generated>)
sunbirdrc-rg-1 | at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
sunbirdrc-rg-1 | at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:688)
sunbirdrc-rg-1 | at dev.sunbirdrc.registry.helper.RegistryHelper$$EnhancerBySpringCGLIB$$eb9a6bdb.authorizeManageEntity(<generated>)
sunbirdrc-rg-1 | at dev.sunbirdrc.registry.controller.RegistryEntityController.postEntity(RegistryEntityController.java:232)
sunbirdrc-rg-1 | at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
sunbirdrc-rg-1 | at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
sunbirdrc-rg-1 | at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
sunbirdrc-rg-1 | at java.lang.reflect.Method.invoke(Method.java:498)
sunbirdrc-rg-1 | at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:190)
sunbirdrc-rg-1 | at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:138)
sunbirdrc-rg-1 | at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:105)
sunbirdrc-rg-1 | at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:878)
sunbirdrc-rg-1 | at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:792)
sunbirdrc-rg-1 | at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
sunbirdrc-rg-1 | at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1040)
sunbirdrc-rg-1 | at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:943)
sunbirdrc-rg-1 | at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
sunbirdrc-rg-1 | at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:909)
sunbirdrc-rg-1 | at javax.servlet.http.HttpServlet.service(HttpServlet.java:652)
sunbirdrc-rg-1 | at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
sunbirdrc-rg-1 | at javax.servlet.http.HttpServlet.service(HttpServlet.java:733)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:227)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
sunbirdrc-rg-1 | at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
sunbirdrc-rg-1 | at org.keycloak.adapters.springsecurity.filter.KeycloakAuthenticatedActionsFilter.doFilter(KeycloakAuthenticatedActionsFilter.java:57)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
sunbirdrc-rg-1 | at org.keycloak.adapters.springsecurity.filter.KeycloakSecurityContextRequestFilter.doFilter(KeycloakSecurityContextRequestFilter.java:61)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
sunbirdrc-rg-1 | at org.keycloak.adapters.springsecurity.filter.KeycloakPreAuthActionsFilter.doFilter(KeycloakPreAuthActionsFilter.java:96)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
sunbirdrc-rg-1 | at org.keycloak.adapters.springsecurity.filter.KeycloakAuthenticationProcessingFilter.successfulAuthentication(KeycloakAuthenticationProcessingFilter.java:214)
sunbirdrc-rg-1 | at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:240)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
sunbirdrc-rg-1 | at org.springframework.web.filter.CorsFilter.doFilterInternal(CorsFilter.java:92)
sunbirdrc-rg-1 | at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
sunbirdrc-rg-1 | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:320)
sunbirdrc-rg-1 | at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:126)
sunbirdrc-rg-1 | at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:90)
sunbirdrc-rg-1 | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
sunbirdrc-rg-1 | at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:118)
sunbirdrc-rg-1 | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
sunbirdrc-rg-1 | at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:137)
sunbirdrc-rg-1 | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
sunbirdrc-rg-1 | at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:111)
sunbirdrc-rg-1 | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
sunbirdrc-rg-1 | at org.keycloak.adapters.springsecurity.filter.KeycloakAuthenticatedActionsFilter.doFilter(KeycloakAuthenticatedActionsFilter.java:74)
sunbirdrc-rg-1 | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
sunbirdrc-rg-1 | at org.keycloak.adapters.springsecurity.filter.KeycloakSecurityContextRequestFilter.doFilter(KeycloakSecurityContextRequestFilter.java:92)
sunbirdrc-rg-1 | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
sunbirdrc-rg-1 | at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:158)
sunbirdrc-rg-1 | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
sunbirdrc-rg-1 | at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63)
sunbirdrc-rg-1 | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
sunbirdrc-rg-1 | at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:116)
sunbirdrc-rg-1 | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
sunbirdrc-rg-1 | at org.keycloak.adapters.springsecurity.filter.KeycloakAuthenticationProcessingFilter.successfulAuthentication(KeycloakAuthenticationProcessingFilter.java:214)
sunbirdrc-rg-1 | at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:240)
sunbirdrc-rg-1 | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
sunbirdrc-rg-1 | at org.keycloak.adapters.springsecurity.filter.KeycloakPreAuthActionsFilter.doFilter(KeycloakPreAuthActionsFilter.java:96)
sunbirdrc-rg-1 | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
sunbirdrc-rg-1 | at org.springframework.security.web.header.HeaderWriterFilter.doHeadersAfter(HeaderWriterFilter.java:92)
sunbirdrc-rg-1 | at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:77)
sunbirdrc-rg-1 | at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
sunbirdrc-rg-1 | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
sunbirdrc-rg-1 | at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:105)
sunbirdrc-rg-1 | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
sunbirdrc-rg-1 | at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:56)
sunbirdrc-rg-1 | at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
sunbirdrc-rg-1 | at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:334)
sunbirdrc-rg-1 | at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:215)
sunbirdrc-rg-1 | at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:178)
sunbirdrc-rg-1 | at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:358)
sunbirdrc-rg-1 | at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:271)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
sunbirdrc-rg-1 | at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
sunbirdrc-rg-1 | at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
sunbirdrc-rg-1 | at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
sunbirdrc-rg-1 | at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
sunbirdrc-rg-1 | at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
sunbirdrc-rg-1 | at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
sunbirdrc-rg-1 | at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
sunbirdrc-rg-1 | at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
sunbirdrc-rg-1 | at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
sunbirdrc-rg-1 | at org.keycloak.adapters.tomcat.AbstractAuthenticatedActionsValve.invoke(AbstractAuthenticatedActionsValve.java:67)
sunbirdrc-rg-1 | at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:542)
sunbirdrc-rg-1 | at org.keycloak.adapters.tomcat.AbstractKeycloakAuthenticatorValve.invoke(AbstractKeycloakAuthenticatorValve.java:181)
sunbirdrc-rg-1 | at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:143)
sunbirdrc-rg-1 | at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
sunbirdrc-rg-1 | at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
sunbirdrc-rg-1 | at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:357)
sunbirdrc-rg-1 | at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:374)
sunbirdrc-rg-1 | at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
sunbirdrc-rg-1 | at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:893)
sunbirdrc-rg-1 | at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1707)
sunbirdrc-rg-1 | at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
sunbirdrc-rg-1 | at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
sunbirdrc-rg-1 | at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
sunbirdrc-rg-1 | at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
sunbirdrc-rg-1 | at java.lang.Thread.run(Thread.java:748)
sunbirdrc-kc-1 | 10:10:34,636 WARN  [org.hibernate.engine.jdbc.spi.SqlExceptionHelper] (default task-2) SQL Error: 0, SQLState: 23505
sunbirdrc-kc-1 | 10:10:34,637 ERROR [org.hibernate.engine.jdbc.spi.SqlExceptionHelper] (default task-2) ERROR: duplicate key value violates unique constraint "sibling_names"
sunbirdrc-kc-1 |   Detail: Key (realm_id, parent_group, name)=(sunbird-rc,  , Issuer) already exists.
`

docker-compose.yaml
`version: '2.4'

services:
  es:
    image: docker.elastic.co/elasticsearch/elasticsearch:7.10.1
    environment:
      - discovery.type=single-node
      - 'ES_JAVA_OPTS=-Xms512m -Xmx512m'
    healthcheck:
      test: ['CMD', 'curl', '-f', 'localhost:9200/_cluster/health']
      interval: 30s
      timeout: 10s
      retries: 4
    ports:
      - '9200:9200'
      - '9300:9300'
  db:
    image: postgres
    ports:
      - '5432:5432'
    environment:
      - POSTGRES_DB=registry
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
  rg:
    image: dockerhub/sunbird-rc-core
    volumes:
      - ./config/schemas:/home/sunbirdrc/config/public/_schemas
    environment:
      - connectionInfo_uri=jdbc:postgresql://db:5432/registry
      - connectionInfo_username=postgres
      - connectionInfo_password=postgres
      - elastic_search_connection_url=es:9200
      - search_provider=dev.sunbirdrc.registry.service.ElasticSearchService
      - sunbird_sso_realm=sunbird-rc
      - sunbird_sso_url=http://kc:8080/auth
      - sunbird_sso_admin_client_id=admin-api
      - sunbird_sso_client_id=registry-frontend
      - sunbird_sso_admin_client_secret=9ab6ffa9-d3fc-40cb-a4b5-0785fcb4ec71
      - sunbird_keycloak_user_set_password=true
      - sunbird_keycloak_user_password=test
      - claims_url=http://cs:8082
    ports:
      - '8081:8081'
    depends_on:
      es:
        condition: service_healthy
      db:
        condition: service_started
      kc:
        condition: service_healthy
  kc:
    image: dockerhub/ndear-keycloak
    volumes:
      - ./config/keycloak:/opt/jboss/keycloak/imports
    environment:
      - DB_VENDOR=postgres
      - DB_ADDR=db
      - DB_PORT=5432
      - DB_DATABASE=registry
      - DB_USER=postgres
      - DB_PASSWORD=postgres
      - KEYCLOAK_USER=admin
      - KEYCLOAK_PASSWORD=admin
      - KEYCLOAK_IMPORT=/opt/jboss/keycloak/imports/realm-export.json
      - KEYCLOAK_FRONTEND_URL=http://localhost:8080/auth
    healthcheck:
      test: ['CMD-SHELL', 'curl -f http://localhost:9990/ || exit 1']
      interval: 30s
      timeout: 10s
      retries: 5
    ports:
      - '8080:8080'
      - '9990:9990'
    depends_on:
      db:
        condition: service_started
  cs:
    image: dockerhub/open-saber-claim-ms
    environment:
      - connectionInfo_uri=jdbc:postgresql://db:5432/registry
      - connectionInfo_username=postgres
      - connectionInfo_password=postgres
      - sunbirdrc_url=http://rg:8081
    ports:
      - '8082:8082'
    depends_on:
      db:
        condition: service_started
      rg:
        condition: service_started
`

**Reply by sachinsahu731** [reply]
@tejash-jl  Any update on this .?

**Reply by tejash-jl** [reply]
I think this is relevant to some issues on roles added in keycloak, can you join [Discord Channel](https://discord.gg/Q5mvw2mGC8) You can ping me there and we can get on a call to resolve this.

**Comment by aniket-tarento** [comment]
@tejash-jl we are not able to start following containers :
1. certificate-api-1 error : `/scripts/docker-entrypoint.sh: line 10: syntax error: unexpected end of file (expecting "then")`
2. keycloak-1
error : ```
09:42:57,964 ERROR [org.jboss.as.controller.management-operation] (ServerService Thread Pool -- 21) WFLYCTL0403: Unexpected failure during execution of the following operation(s): [{
    "operation" => "add",
    "address" => [("subsystem" => "request-controller")]
}]: java.lang.RuntimeException: WFLYCTL0195: Interrupted awaiting transaction commit or rollback
        at org.jboss.as.controller@15.0.1.Final//org.jboss.as.controller.ParallelBootOperationStepHandler$ParallelBootTransactionControl.operationPrepared(ParallelBootOperationStepHandler.java:458)
        at org.jboss.as.controller@15.0.1.Final//org.jboss.as.controller.ModelController$OperationTransactionControl.operationPrepared(ModelController.java:131)
        at org.jboss.as.controller@15.0.1.Final//org.jboss.as.controller.AbstractOperationContext.executeDoneStage(AbstractOperationContext.java:874)
        at org.jboss.as.controller@15.0.1.Final//org.jboss.as.controller.AbstractOperationContext.processStages(AbstractOperationContext.java:805)
        at org.jboss.as.controller@15.0.1.Final//org.jboss.as.controller.AbstractOperationContext.executeOperation(AbstractOperationContext.java:468)
        at org.jboss.as.controller@15.0.1.Final//org.jboss.as.controller.ParallelBootOperationStepHandler$ParallelBootTask.run(ParallelBootOperationStepHandler.java:384)
        at org.jboss.threads@2.4.0.Final//org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1990)
        at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
        at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1377)
        at java.base/java.lang.Thread.run(Thread.java:829)
        at org.jboss.threads@2.4.0.Final//org.jboss.threads.JBossThread.run(JBossThread.java:513)
```

**Reply by tejash-jl** [reply]
@aniket-tarento please share the release version of the containers used.
